overlord/snapstate/autorefresh_gating.go

```diff
@@ -517,15 +518,22 @@ func createGateAutoRefreshHooks(st *state.State, affectedSnaps map[string]*affec
 // snapsToRefresh returns all snaps that should proceed with refresh considering
 // hold behavior.
 var snapsToRefresh = func(gatingTask *state.Task) ([]*refreshCandidate, error) {
-	// TODO: consider holding (responses from gating hooks) here.
-	// Return all refresh candidates for now.
 	var snaps map[string]*refreshCandidate
 	if err := gatingTask.Get("snaps", &snaps); err != nil {
 		return nil, err
 	}
+
+	held, err := heldSnaps(gatingTask.State())
+	if err != nil {
+		return nil, err
+	}
 	var candidates []*refreshCandidate
 	for _, s := range snaps {
-		candidates = append(candidates, s)
+		if !held[s.InstanceName()] {
+			candidates = append(candidates, s)
+		} else {
+			logger.Noticef("skipping refresh of snap %q", s.InstanceName())
```

Comment:
this is a bit verbose, can't we accumulate the names on a list, and then do a single log?

also s/snap/held snap/